megatron/model/language_model.py

```diff
@@ -312,7 +312,7 @@ def __init__(self,
                  add_pooler=False,
                  pre_process=True,
                  post_process=True,
-                 num_experts=1):
+                 num_experts=[1]):
```

Comment:
why do we need a list here?

Reply:
Because in order to support pyramid MoE we later changed args.num_experts to list type https://github.com/microsoft/Megatron-DeepSpeed/blob/b4d4a0e06577218a21abf18eb7d8088bb8ecaad3/megatron/arguments.py#L288. Some num_experts defaults in model code already changed to list (e.g, https://github.com/microsoft/Megatron-DeepSpeed/blob/b4d4a0e06577218a21abf18eb7d8088bb8ecaad3/megatron/model/transformer.py#L612), but not in this language_model.py.

Reply:
Got it. Thanks for catching it. I am a little surprised we didn't get any errors in running gpt.